### PR TITLE
Print warning if JDK version is older than 11

### LINKED ISSUE #7730: Print deprecation warning when Java 8 is in use
**Description**
As per #5691, we should make the deprecation of Java 8 official by printing a (preferably large) warning when Java 8 is used for application augmentation.

### PR DESCRIPTION
**Pull request by dmlloyd**
Fixes #7730

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkus/deployment/QuarkusAugmentor.java [hunk 1]
```diff
@@ -74,6 +76,12 @@
     }
 
     public BuildResult run() throws Exception {
+        Pattern pattern = Pattern.compile("(?:1\\.)?(\\d+)(?:\\..*)?");
+        Matcher matcher = pattern.matcher(System.getProperty("java.version", ""));
+        if (matcher.matches() && Integer.parseInt(matcher.group(1)) < 11) {
+            log.warn("Using Java versions older than 11 to build"
+                    + " Quarkus applications is deprecated and will be disallowed in a future release!");
```
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/deployment/QuarkusAugmentor.java]: I usually refrain from using exclamation marks in user messages but feel free to ignore.

**Review by gsmet — APPROVED** [review]

File: core/deployment/src/main/java/io/quarkus/deployment/QuarkusAugmentor.java (hunk 1 above)
Comment (dmlloyd) [reply core/deployment/src/main/java/io/quarkus/deployment/QuarkusAugmentor.java]: TBH I was wondering if it should be bigger... like, a big banner

```
***********************************************************************
*** WARNING WARNING WARNING WARNING WARNING WARNING
*** 🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥🔥
***********************************************************************

```

Comment (Sanne) [reply core/deployment/src/main/java/io/quarkus/deployment/QuarkusAugmentor.java]: with animated, red flames :)

Comment (gastaldi) [reply core/deployment/src/main/java/io/quarkus/deployment/QuarkusAugmentor.java]:
```
/ Using Java versions older than 11 to   \
| build Quarkus applications is          |
| deprecated and will be disallowed in a |
\ future release. I mean, moo            /
 ----------------------------------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||
```

Comment (gsmet) [reply core/deployment/src/main/java/io/quarkus/deployment/QuarkusAugmentor.java]: Ok, well, I will merge the version with the exclamation mark before someone comes with more brilliant ideas :).

**Pull request merged**